### PR TITLE
fix(learn): warn about XSS in convert JSON to HTML

### LINKED ISSUE #36162: Challenges use innerHTML without mentioning XSS vulnerabilities
Several challenges from [here](https://learn.freecodecamp.org/data-visualization/json-apis-and-ajax/get-json-with-the-javascript-xmlhttprequest-method/) to [here](https://learn.freecodecamp.org/data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method/) use `innerHTML` to update the page.  These are the first places where this technique is introduced, but don't mention any security concerns.

Since the first two challenges use `innerHTML` just to insert text, they could use `textContent` instead.  Later challenges insert actual html and would be a good place to mention the possibility of XSS.  They should have a warning about XSS and maybe include a sanitizer function to hammer home the point.  The warning would not have to go into depth, a note and a link to the [guide](https://guide.freecodecamp.org/security/xss-cross-site-scripting/) would be good.

How does this sound?

### PR DESCRIPTION
**Pull request by Wocanilo**
- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [X] My changes do not use shortened URLs or affiliate links.

Related to #36162

### CODE REVIEW

**Comment by RandellDawson** [comment]
@Wocanilo The same verbiage as the English changes you suggested in #37036 will apply here as well.

File: curriculum/challenges/spanish/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.spanish.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ localeTitle: Convertir datos JSON a HTML
...
+<b>Nota:</b> Para completar este reto es necesario añadir nuevos elementos html, por lo que no podrá depender de textContent. Deberá utilizar el método innerHTML que podría hacer que su página web sea vulnerable a ataques de cross site scripting.
```
Comment (RandellDawson) [inline curriculum/challenges/spanish/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.spanish.md]: Something I did not catch in the first review, but according to our guidelines, notes are surrounded by `strong` tags.
```suggestion
<strong>Nota:</strong> Para completar este reto es necesario añadir nuevos elementos html, por lo que no podrá depender de textContent. Deberá utilizar el método innerHTML que podría hacer que su página web sea vulnerable a ataques de cross site scripting.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
One more thing to fix.

**Review by batistaDev1113 — CHANGES_REQUESTED** [review]
I think the following term should be changed from **Cuando el bucle termina** to **Cuando la repetición de instrucciones termina**

**Comment by RandellDawson** [comment]
@batistaDev1113 Since your suggested change is not part of the document which the user is changing, can you please create a separate PR to make your suggested change?

Thank you.

**Comment by moT01** [comment]
I made the suggested changes to this, that were suggested on the linked PR. It should be good to go.

**Pull request merged**